bundles/binding/org.openhab.binding.tacmi/src/main/java/org/openhab/binding/tacmi/internal/TACmiMeasureType.java

```diff
@@ -9,44 +9,84 @@
 package org.openhab.binding.tacmi.internal;
 
 import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /**
  * This enum holds all the different measures and states available to be
- * retrieved by the TACmi binding Not the complete list yet.
- * 
- * @author Timo Wendt
+ * retrieved by the TACmi binding, including the scale factors needed to convert the received values to the real
+ * numbers.
+ *
+ * @author Timo Wendt, Wolfgang Klimt
  * @since 1.8.0
  */
 public enum TACmiMeasureType {
-    NONE(0),
-    TEMPERATURE(1),
-    SECONDS(4),
-    UNSUPPORTED(-1);
+    NONE(0, 0),
+    TEMPERATURE(1, 1),
+    UNKNOWN2(2, 0),
+    UNKNOWN3(3, 0),
+    SECONDS(4, 0),
+    UNKNOWN5(5, 0),
+    UNKNOWN6(6, 0),
+    UNKNOWN7(7, 0),
+    UNKNOWN8(8, 0),
+    UNKNOWN9(9, 0),
+    KILOWATT(10, 1),
+    KILOWATTHOURS(11, 1),
+    MEGAWATTHOURS(12, 0),
+    UNKNOWN13(13, 0),
+    UNKNOWN14(14, 0),
+    UNKNOWN15(15, 0),
+    UNKNOWN16(16, 0),
+    UNKNOWN17(17, 0),
+    UNKNOWN18(18, 0),
+    UNKNOWN19(19, 0),
+    UNKNOWN20(20, 0),
+    UNKNOWN21(21, 0),
+
+    UNSUPPORTED(-1, 0);
+
+    private int measure;
+    private int scale;
 
-    int measure;
+    private static final Logger logger = LoggerFactory.getLogger(TACmiMeasureType.class);
 
-    private TACmiMeasureType(int measure) {
+    private TACmiMeasureType(int measure, int scale) {
         this.measure = measure;
+        this.scale = scale;
     }
 
     public int getMeasure() {
         return measure;
     }
 
+    public int getScale() {
+        return scale;
+    }
+
     /**
      * Return the measure type for the specified name.
-     * 
+     *
      * @param measure
      * @return
      */
     public static TACmiMeasureType fromString(String measure) {
         if (!StringUtils.isEmpty(measure)) {
-            for (TACmiMeasureType measureType : TACmiMeasureType.values()) {
-                if (measureType.toString().equalsIgnoreCase(measure)) {
-                    return measureType;
-                }
-            }
+            return TACmiMeasureType.valueOf(measure.toUpperCase());
         }
         throw new IllegalArgumentException("Invalid measure: " + measure);
     }
+
+    /**
+     * Return measure type for a specific int value
+     */
+    public static TACmiMeasureType fromInt(int type) {
+        for (TACmiMeasureType mtype : TACmiMeasureType.values()) {
+            if (mtype.ordinal() == type) {
```

Comment:
Would it not be better to use `getMeasure()` instead of `ordinal()` because you don't want to depend on the measure types being the same as the ordinal index of their enum constants?